reference/reference_test.go

```diff
@@ -96,6 +96,23 @@ func TestReferenceParse(t *testing.T) {
 			err:   digest.ErrDigestUnsupported,
 		},
 		{
+			input: "Uppercase:tag",
+			err:   ErrNameContainsUppercase,
+		},
+		{
+			input: "Uppercase/lowercase:tag",
+			err:   ErrNameContainsUppercase,
```

Comment:
Looks like there's a bug in the regexp; this reference fails in docker, but passes here;

```
--- FAIL: TestReferenceParse (0.00s)
       	reference_test.go:173: "Uppercase/lowercase:tag": missing expected error: repository name must lowercase
FAIL
```

But in docker;

```
docker tag busybox UpperCase/lowercase:tag

Error response from daemon: invalid reference format: repository name must be lowercase
```

I suspect the regex here treats the first component as a domain-name (which is allowed to have uppercase characters), because the test below (which includes a domain-name), *does* pass